icd/cache_support/binary_cache_serialization.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Google LLC. All Rights Reserved.
```

Comment:
Please check if you are fine with this copyright notice.

Reply:
@jinjianrong probably knows what we should be doing for the copyright header on new files.